### PR TITLE
Install SUSEConnect before using it

### PR DESCRIPTION
**Pull request by drpaneas**
- Related ticket: https://progress.opensuse.org/issues/34768
- Needles: *not needed*
- Verification run: http://skyrim.qam.suse.de/tests/2664#

### CODE REVIEW

File: lib/registration.pm [hunk 1]
```diff
@@ -620,6 +620,7 @@ sub install_docker_when_needed {
             assert_script_run('zypper se docker || zypper -n ar -f http://download.suse.de/ibs/SUSE:/SLE-12:/Update/standard/SUSE:SLE-12:Update.repo');
         }
         else {
+            zypper_call('in SUSEConnect');
```
Comment (asmorodskyi) [inline lib/registration.pm]: probably better to insert it inside `add_suseconnect_product` so it will fix problem in more generic way ?

**Comment by okurz** [comment]
I'm sorry but I think there is something you completely misunderstood about our products. Why on earth should we need SUSEConnect on openSUSE Leap 15.0 to install docker? And if that *would* be necessary on SLE15, where would you get the SUSEConnect package from when it is not registered? Please do not merge this PR as IMHO it is completely invalid. Please see my comments in the progress ticket instead.

The problem is in the logical expression: We have `is_caasp`, else we have `is_sle('<15')` and here we end up in the else-branch.

Yes, I broke it with https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/4805 . Sorry about that.

I will fix it.

**Comment by okurz** [comment]
https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/4831

**Comment by drpaneas** [comment]
@okurz it was a misunderstanding.

Apparently I've got confused by https://openqa.opensuse.org/tests/654843#step/docker/2 -- I really hate these hacks with *is_xyz*, mostly because they are misleading, like in this case. I actually didn't even touch that part of the logical expression in my commit; the error was already merged in *master branch*.

Anyway, thank you once again for spotting the problem and going the extra mile by fixing it ;)

**Comment by okurz** [comment]
yeah I was concerned about this confusion already when the docker install function was moved into lib/registration by @mnowaksuse in d0addb4e. Please see my proposal how to change that in https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/4842

**Pull request closed without merging**